src/store/preferences.ts

```diff
@@ -206,6 +206,22 @@ export const actions = {
 
     return severities;
   },
+  async toggleShowMuted({ dispatch, rootState }: PrefActionContext, isMuted: boolean) {
+    await dispatch(
+      'preferences/updatePreferencesData',
+      {
+        property: 'diagnostics.showMuted',
+        value:    isMuted,
+      },
+      { root: true },
+    );
+
+    await dispatch(
+      'preferences/commitPreferences',
```

Comment:
Will this accidentally commit other, unrelated changes? For example, if I also have the prefs window open and managed to change some things…
(I guess that's impossible due to the prefs window being modal?)

Reply:
This is a very good catch. I was originally trying to avoid changing `commitPreferences` but I think it makes sense that we do so in order to avoid this. 

I've added an optional `payload` parameter that we can use to update only the props that interest us instead of the entirety of the preferences state.